### ISSUE TITLE
Pipeline tasks are queued up but don't start

### ISSUE THREAD

**Issue by suibinz** [siamaksade/openshift-cd-demo#43]
either trigger from Jenkins or from oc build, the pipeline tasks are queued up but none would start running on Jenkins.
The build jobs are stuck at "No stages have started."

**Comment by bonocore** [comment]
Happened also to me, not sure about the real cause (proably lack of resources on my laptop).

As a workaround, I've find out that cleaning up everything and starting from scratch is a solution.

The script I have used to cleanup is something like:

oc project cicd 
oc delete all -l template=cicd -n cicd --now=true
oc delete all -l app=shopping -n dev --now=true
oc delete template cicd -n cicd
oc delete all -l app=shopping -n stage --now=true
oc delete rolebinding default_edit -n cicd
oc delete configmap gogs-install
oc delete is tasks -n dev
oc delete is tasks -n stage
oc delete template cicd

**Comment by suibinz** [comment]
Yes, the behavior is somewhat intermittent - I can start up the template from fresh, sometime works, sometime not.  It is probably not a resource issue. I am running an m4.large on EC2, and CPU  and mem are  not heavily utilized. 

There might be race condition on the sequence of services. Similar problem of Gogs service initiated before the "install-gogs" task is done. I think the cicd template should be StatefulSet instead.  I am new to this, but am looking into modifying it.
